splunk_add_on_ucc_framework/commands/rest_builder/endpoint/base.py

```diff
@@ -33,10 +33,14 @@
 class RestEntityBuilder:
     _title_template = "[{}]"
     _rh_template = """
+special_fields = [
+{special_fields}
+]
+
```

Comment:
Continuing from #1343, the logic of not having an emtpy list is:
1. We have mentioned this parameter as optional in splunktaucclib for `RestModel` class. Or, if we are always passing this object to `RestModel` class, perhaps it should have this parameter as required instead of optional.
2. Declaring a variable when we know it is empty seems illogical to just ship. And going down the line, this introduces tech debt to be cleaned up.

Reply:
ad 1. The optional parameter was set in splunktaucclib because we want it to be backward compatible. There may be other projects that use splunktaucclib and making it required would force us to make a breaking change.
ad 2. Special fields are currently used to collect all "name" fields. From what I've checked, this is a mandatory field in the schema and I don't see a situation where this list would be empty, but I'll refactor this as an optional parameter in UCC as well.